crates/biome_string_case/src/lib.rs

```diff
@@ -445,71 +445,48 @@ pub trait Collator {
     ) -> Ordering {
         let mut iter1 = iter1.into_iter();
         let mut iter2 = iter2.into_iter();
-        loop {
-            match (iter1.next(), iter2.next()) {
-                (Some(c1), Some(c2)) if c1 == c2 => {}
```

Comment:
Third bug

Reply:
I think that this code is right. The `loop`and this empty block are important: we are comparing strings, you have to continue the comparison while the characters are equal.